### DISCUSSION TITLE
Does app on Android can be fullscreen without status bar ?

### DISCUSSION THREAD

**Discussion by 18o** [tauri-apps/tauri#9261] [Q&A]
Hi, i'm working with tauri 2 for building android app. 
but didn't find a way to set the Activity fullscreen and how to hide the status bar ?
does anyone can help me? even just some idea is very thanks.

**Comment by MatthewZenn** [comment]
Were you able to find an answer to this?

**Comment by FabianLars** [comment]
Changing the MainActivity.ks file to this:
```kt
package // Keep the old value here

import android.os.Bundle
import androidx.core.view.WindowCompat

class MainActivity : TauriActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    WindowCompat.setDecorFitsSystemWindows(window, false)
  }
}
```
and adding this to the theme file (i forgot its path)
```xml
<item name="android:statusBarColor" tools:targetApi="l">@android:color/transparent</item>
```
should _mostly_ work.

A problem with this is opening the keyboard doesn't shift the layout so if you have input fields at the bottom half of the screen they won't be visibile anymore. I did not look into how to best solve this yet.

**Reply by MatthewZenn** [reply]
If not doing the code that fits the screen size, you can just replace the '@android:color/transparent' part with your own hexcode, to change the color of the statusbar.

**Reply by Burial0268** [reply]
still working on the latest rc version?

**Reply by FabianLars** [reply]
i think it's still working, yes. i did not hear any reports saying otherwise.

**Reply by alexiokay** [reply]
for me it doesn't work. I found some solution, but when I change phone orientation then I see black bar on the side ... in MainActivity.kt: Rather:  

`` package com.tauri.dev

import android.os.Bundle
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

class MainActivity : TauriActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Allow content to extend under the system bars
        WindowCompat.setDecorFitsSystemWindows(window, false)

        // Get the insets controller to manage system UI
        val windowInsetsController = WindowCompat.getInsetsController(window, window.decorView)


        // Ensure the status bar is fully transparent
        window.statusBarColor = android.graphics.Color.TRANSPARENT
    }
`}`

**Reply by little-starts** [reply]
> for me it doesn't work. I found some solution, but when I change phone orientation then I see black bar on the side ... in MainActivity.kt: Rather:
> 
> `` package com.tauri.dev
> 
> import android.os.Bundle import androidx.core.view.WindowCompat import androidx.core.view.WindowInsetsCompat import androidx.core.view.WindowInsetsControllerCompat
> 
> class MainActivity : TauriActivity() { override fun onCreate(savedInstanceState: Bundle?) { super.onCreate(savedInstanceState)
> 
> ```
>     // Allow content to extend under the system bars
>     WindowCompat.setDecorFitsSystemWindows(window, false)
> 
>     // Get the insets controller to manage system UI
>     val windowInsetsController = WindowCompat.getInsetsController(window, window.decorView)
> 
> 
>      Ensure the status bar is fully transparent
>     window.statusBarColor = android.graphics.Color.TRANSPARENT
> }
> ```
> 
> `}`

Thank you very much for the code, I think I've tried out the final solution, this works on my device, I added the ability to hide all status bars, navigation bars, screen notches, and tested that it works in my device
```kt
package // Keep your package name
import android.os.Bundle
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

class MainActivity : TauriActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Allow content to extend under the system bars
        WindowCompat.setDecorFitsSystemWindows(window, false)

        // Get the insets controller to manage system UI
        val windowInsetsController = WindowCompat.getInsetsController(window, window.decorView)

        // Hide both status bar and navigation bar
        windowInsetsController.apply {
            hide(WindowInsetsCompat.Type.systemBars())
            systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        }

        // Make both bars transparent
        window.statusBarColor = android.graphics.Color.TRANSPARENT
        window.navigationBarColor = android.graphics.Color.TRANSPARENT

        // Handle display cutout (notch)
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
            window.attributes.layoutInDisplayCutoutMode = 
                android.view.WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES
        }
    }
}
```

**Reply by alexiokay** [reply]
Thanks, I found it already too, just forgot to post it here. Good job.

**Comment by thiagovianav** [comment]
is there a way to use it like a function? my app doesn't need all the time, just in reader mode

**Comment by saurL** [comment]
for anyone wondering if you add 
```java
ViewCompat.setOnApplyWindowInsetsListener(findViewById(android.R.id.content)) { view: View, insets: WindowInsetsCompat ->
            val bottomInset = insets.getInsets(WindowInsetsCompat.Type.ime()).bottom
            view.setPadding(0, 0, 0, bottomInset)
            insets
        }
```

in `override fun onCreate(savedInstanceState: Bundle?) {` it fix the keyboard issue.

**Comment by SleekGeek-254** [comment]
i dont know if anyone is still looking for this answer the Clean way to do it is to add the `enableEdgeToEdge` import and implement full-screen mode:

## Current State
Your `MainActivity.kt` is very simple:
```kotlin
package com.XXXX.app

class MainActivity : TauriActivity()
```

## Proposed Changes

To enable edge-to-edge (full screen) display, you need to:

1. **Add the import statement** at the top:
   ```kotlin
   import androidx.activity.enableEdgeToEdge
   ```

2. **Override the `onCreate` method** and call `enableEdgeToEdge()` before calling `super.onCreate()`:
   ```kotlin
   override fun onCreate(savedInstanceState: Bundle?) {
       enableEdgeToEdge()
       super.onCreate(savedInstanceState)
   }
   ```

3. **Add the necessary Android import** for `Bundle`:
   ```kotlin
   import android.os.Bundle
   ```

## Final Result
The complete file should look like this:

```kotlin
package com.XXXX.app

import android.os.Bundle
import androidx.activity.enableEdgeToEdge

class MainActivity : TauriActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge()
        super.onCreate(savedInstanceState)
    }
}
```

4. Add `implementation("androidx.activity:activity-ktx:1.10.1")` to the dependencies in `build.gradle.kts`


## Important Notes
- `enableEdgeToEdge()` must be called **before** `super.onCreate()` for it to work properly
- This will make your app draw content under the system bars (status bar and navigation bar)
- You may need to handle window insets in your UI to prevent content from being obscured by system bars
- This requires `androidx.activity:activity` version 1.6.0 or higher in your dependencies
